internal/transport/http_util.go

```diff
@@ -335,10 +335,12 @@ func (w *bufWriter) Write(b []byte) (n int, err error) {
 		w.offset += nn
 		n += nn
 		if w.offset >= w.batchSize {
-			err = w.flushKeepBuffer()
+			if err = w.flushKeepBuffer(); err != nil {
```

Comment:
The problem with not returning the number of bytes written from `flushKeepBuffer` is that we can now return a wrong value for the number of bytes written from this function.

The value `n` has already been incremented in line 336. But we don't know how much of it was actually written.

So, I would recommend that you fix `flushKeepBuffer` to return the number of bytes, and update the value of `n` here appropriately, and thus guarantee that the value we return from this function is accurate.

Reply:
@easwars updated